dis_snek/client.py

```diff
@@ -462,6 +464,46 @@ def wait_for(self, event: str, checks: Optional[Callable[..., bool]] = MISSING,
 
         return asyncio.wait_for(future, timeout)
 
+    async def wait_for_component(
+        self,
+        messages: Union[Message, int, list] = None,
+        components: Union[str, dict, list] = None,
+        check=None,
+        timeout=None,
+    ) -> Component:
```

Comment:
it returns an awaitable future, not component object.

Reply:
How do I typehint that? `Awaitable[Future]` doesn't work since Future isn't in typing, so I just removed the typehint for now

Reply:
Nevermind, I added `from asyncio import Future` into `if TYPE_CHECKING:` and then used typehint `Awaitable['Future']`